### DISCUSSION TITLE
FieldHub Local Machine Server

### DISCUSSION THREAD

**Discussion by matidav** [dainst/idai-field-client#174] [Q&A] [answered]
Hi, I'm trying to test FieldHub server on my local machine and I've been following the deployment example. I've set up the docker container and set my environment variable HOST=localhost. Both containers are working as expected (i.e. I am able to access the fieldhub app and the Couchdb database), but I am unable to synchronize my Field Desktop with the locally running server. In the "Download project" tab of Field Desktop, I've put in localhost, http://localhost:80, http://127.0.0.1:80, http://localhost:4000, etc. but nothing has seemed to work. After I have set up the server, how do I synchronize it with Field Desktop?

**Comment by tkleinke** [comment, marked as answer]
Hi! The correct address is http://localhost:80 (or just http://localhost). The problem is probably that you haven't created the project in the Desktop application yet. You can only download projects that have been created via the menu Project -> New... and synchronized to the FieldHub server.

This should work:
1. Create a project in the FieldHub interface (accessible in the browser via http://localhost)
2. Open Field Desktop, select the menu Project -> New..., and create a project of the same name.
3. Open the menu Project -> Synchronize..., enter http://localhost as address and the password chosen in the FieldHub interface. Enable the switch "Activate synchronization" and apply the settings.
4. The project is now available in FieldHub and can be downloaded to other computers using Field Desktop.

**Reply by matidav** [reply]
Thank you! This worked! I assume that the "Host" environment variable is for communication between the fieldhub app and the couch db database and the variable should match url where the fieldhub app is being deployed? Also, if you have any tips for deploying the server on AWS or other cloud hosting services, I would really appreciate it!

**Reply by tkleinke** [reply]
Glad to hear it's working now! :)

Yes, the HOST variable should be set to the domain you want to run the application from. It's used by the Phoenix framework to create the URLs of the FieldHub web application.

I'm afraid we don't have any experience with deploying FieldHub on cloud hosting services yet, at least I'm not aware of it. Maybe @dersmon (mainly responsible for FieldHub development & support) has something to add here, he's currently on vacation and will be back in about two weeks.

**Comment by dersmon** [comment]
> Thank you! This worked! I assume that the "Host" environment variable is for communication between the fieldhub app and the couch db database and the variable should match url where the fieldhub app is being deployed? Also, if you have any tips for deploying the server on AWS or other cloud hosting services, I would really appreciate it!

I have no experience with cloud services as we use our own server infrastructure. 

My initial thoughts: I think running the application itself will probably be trivial using docker. But there are two open questions I would research first:

1. What service to use/how to mount the the image file directory. In general you do not need the fastest file access for FieldHub (you will not have a lot of users and it is a service running in the background), so I would probably value cheap volume over fast access.  
2. How to backup the aforementioned file directory _and_ CouchDB data regularly.

Those problems will 100% be solved already, I just do not know how.

If you come to a decision using cloud infrastructure please let us know, I would imagine other people would also be interested.

Cheers,
Simon
